pkg/utils/kubernetes/utils_test.go

```diff
@@ -0,0 +1,18 @@
+package kubernetes
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"testing"
+)
+
+func TestCustomResourceDefinitionExists(t *testing.T) {
```

Comment:
Is this intended for unit testing? Shouldn't it use mock client to work?

Reply:
Mock added

Reply:
This is not how unit testing is done in kuberenetes/openshift. Please see the unit tests for some of the operators, for example those of kie-cloud. There is mock client provided by kubernetes and used (sometimes enhanced by us) in unit tests.

Reply:
TestCustomResourceDefinitionExistsWithMockClient(). This method I used mock objects, 
I don't see comment on it or yet, can I assume it is the right way for the unit test?

Reply:
No, it is not the right way. You're not supposed to make a fake web server to mock the k8s API.

Reply:
In order to test the CustomResourceDefinitionExists func, I have to have a fake rest.Config object which is usually created with a fake server as demonstrated in client-go.rest.fake pkg, and [DiscoveryClient.ServerResourcesForGroupVersion() which I am calling ](https://github.com/kubernetes/client-go/blob/573f9163af8594d7a0df7767101231f3864315ae/discovery/discovery_client.go#L200) needs to connect to an rest.Config to get the APIResourceList, and that is why I am used it.

I checked a couple of operators, don't see anyone is faking the rest.Config.  The one in kie-cloud, like mock_service.go, mainly creating fake client & scheme, but these are not what I need to test the new func.   I do need a discovery client but not the client obj. in operator.

I may misunderstood you comment, could you elaborate more what should be the right mock for this rest.Config?

Reply:
Let's design and get this done based on the work and functionality, and discuss the unit testing later. There are way too many comments, and getting back to this every 24 hours means I no longer remember where we're at.

Reply:
deleted the utils_test.go for now, and the utils.go is ready for review.